mount.go

```diff
@@ -0,0 +1,261 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	pb "github.com/hyperhq/runv/hyperstart/api/grpc"
+	goudev "github.com/jochenvg/go-udev"
```

Comment:
That one is a cgo wrapper around libudev. We can live with that for now, but we have to keep in mind that this makes our binary no longer static and brings glibc, lpthread and libudev dependencies.
And what we need from it is basically a netlink socket and monitor from it, so we probably should consider adding UEVENT support to github.com/vishvananda/netlink/. The genetlink support is already there, so it **should** not be a huge task (famous last words..)

Reply:
I would love that. This would be so much cleaner. I can take a look at this in parallel, so that we can improve pretty quickly.